app/plugin/LineSliceHighlight.js

```diff
@@ -0,0 +1,83 @@
+Ext.define('CpsiMapview.plugin.LineSliceHighlight', {
+    extend: 'Ext.plugin.Abstract',
+    alias: 'plugin.cmv_line_slice_highlight',
+    pluginId: 'cmv_line_slice_highlight',
+
+    /**
+     * @type {ol.StyleFunction|ol.style.Style}
+     */
+    style: null,
+
+    constructor: function () {
+        this.style = new ol.style.Style({
+            stroke: new ol.style.Stroke({
+                color: 'red',
+                width: 2
+            })
+        });
+    },
+
+    /**
+     * Calculates the slice of geometry from start to end
+     * @param {ol.geom.LineString} geometry
+     * @param {number} start in m
+     * @param {number} end in m
+     * @returns {ol.geom.LineString}
+     */
+    calculateSlice: function (geometry, start, end) {
+        var length = 0;
+        var coordinates = [];
+        geometry.forEachSegment(function (a, b) {
+            var segment = new ol.geom.LineString([a, b]);
+            var segmentLength = ol.Sphere.getLength(segment);
+            if (length <= start && start < length + segmentLength) {
+                // start is in this segment
+                coordinates.push(segment.getCoordinateAt((start - length) / segmentLength));
+            }
+            if (start <= length + segmentLength && length + segmentLength <= end) {
+                // the endpoint of the segment is between start and end
+                // openlayers forEachSegment reuses the arrays for the coordinates so it needs to be cloned
+                coordinates.push(b.slice());
+            }
+            if (length <= end && end < length + segmentLength) {
+                // end is in this segment
+                coordinates.push(segment.getCoordinateAt((end - length) / segmentLength));
+            }
+
+            length += segmentLength;
+        });
+        return new ol.geom.LineString(coordinates);
+    },
+
+    highlightSlice: function (geometry, start, end) {
```

Comment:
Could you please add API docs with a short description? Also for `removeHighlight` and `setStyle`.